### PR TITLE
added tests for v0.0.2

### PR DESCRIPTION
**Pull request by christian-forta**

### CODE REVIEW

File: alert-combiner-py/src/constants.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+             "0x4adff9a0ed29396d51ef3b16297070347aab25575f04a4e2bd62ec43ca4508d2",  # money laundering
+             "0x0ffe038c802784f739bb27fcd4274f71c384fea78de87c9ef8d5b3fb72b514c7",  # high gas usage
+             "0xd92be855c14c12cf7ca43315deeaafbee8ec8dc732b0a452c9a6e2165dc554b0",  # high gas price
+             "0xbf953b115fd214e1eb5c4d6f556ea30f0df47bd86bf35ce1fdaeff03dc7df5b7"  # high value transaction
+             ]
+
+ALERT_ID_STAGE_MAPPING = {"FORTA-PHISHING-ALERT": "Preparation",  # ice phishing
+                          "SUSPICIOUS-CONTRACT-CREATION": "Preparation",  # suspicious contract creation
+                          "SUSPICIOUS-CONTRACT-CREATION-TORNADO-CASH": "Preparation",  # suspicious contract creation
+                          "FORTA-BLOCKLIST-ADDR-TX": "Preparation",  # ice phishing
```
Comment (mwakaba2) [inline alert-combiner-py/src/constants.py:19]: nit: `# blocklisted account tx`

File: alert-combiner-py/src/forta_explorer.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+
+class FortaExplorer:
+
+    def empty_alerts(self) -> pd.DataFrame:
+        df_forta = pd.DataFrame(columns=['createdAt', 'name', 'protocol', 'findingType', 'source', 'severity', 'metadata', 'alertId', 'description', 'addresses', 'contracts', 'hash'])
+        return df_forta
+
+    def alerts_by_agent(self, agent_id: str, start_date: datetime, end_date: datetime, results_limit: int = 0) -> pd.DataFrame:
+        count = 1
```
Comment (mwakaba2) [inline alert-combiner-py/src/forta_explorer.py:16]: count re-assignment at line #22

File: alert-combiner-py/src/agent.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+        addresses = set()
+        for index, row in money_laundering.iterrows():
+            addresses = addresses.union(set(row['addresses']))
+
+        # analyze each address' alerts
+        for potential_attacker_address in addresses:
+            logging.debug(potential_attacker_address)
+            # if address is a contract or null address, skip
+            if(is_contract(w3, potential_attacker_address) or potential_attacker_address[0:12] == '0x0000000000'):
```
Comment (mwakaba2) [inline alert-combiner-py/src/agent.py:92]: can also do
`potential_attacker_address.startswith('0x0000000000')`

File: alert-combiner-py/src/agent.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+            stages = set()
+            involved_addresses = set()
+            address_alerts = df_forta_alerts[df_forta_alerts["addresses"].apply(lambda x: potential_attacker_address in x)]
+            involved_alert_ids = address_alerts["alertId"].unique()
+            for alert_id in involved_alert_ids:
+                if alert_id in ALERT_ID_STAGE_MAPPING.keys():
+                    stage = ALERT_ID_STAGE_MAPPING[alert_id]
+                    stages.add(stage)
+                    # get addresses from address field to add to involved_addresses
+                    address_alerts[address_alerts["alertId"] == alert_id]["addresses"].apply(lambda x: involved_addresses.update(set(x)))
```
Comment (mwakaba2) [inline alert-combiner-py/src/agent.py:105]: `apply` will return a new dataframe, and there's no assignment here, so the updated dataframe is not saved.

File: alert-combiner-py/src/forta_explorer.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+import logging
+
+
+class FortaExplorer:
+
+    def empty_alerts(self) -> pd.DataFrame:
+        df_forta = pd.DataFrame(columns=['createdAt', 'name', 'protocol', 'findingType', 'source', 'severity', 'metadata', 'alertId', 'description', 'addresses', 'contracts', 'hash'])
+        return df_forta
+
+    def alerts_by_agent(self, agent_id: str, start_date: datetime, end_date: datetime, results_limit: int = 0) -> pd.DataFrame:
```
Comment (mwakaba2) [inline alert-combiner-py/src/forta_explorer.py:15]: it seems like this parameter is no longer used? `results_limit`

File: alert-combiner-py/src/forta_explorer.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        count = 1
+        url = "https://api.forta.network/graphql"
+
+        df_forta = self.empty_alerts()
+        json_data = ""
+        first_run = True
+        count = 0
+        while (json_data == "" or json_data['data']['alerts']['pageInfo']['hasNextPage']):
+            query = """query exampleQuery {
+                    # first 5 alerts
```
Comment (mwakaba2) [inline alert-combiner-py/src/forta_explorer.py:25]: nit: remove old comment

**Review by mwakaba2 — APPROVED** [review]
just some small nitpicks and one question about [line 105](https://github.com/forta-network/starter-kits/pull/9/files#diff-3c8ea38e5683bc18cdc64e4f58336279e5b71eb1797238defe0d79ff3db257daR105)

File: alert-combiner-py/src/agent.py (hunk 4 above)
Comment (christian-forta) [reply alert-combiner-py/src/agent.py:105]: Mariko, it is as the update operates on the involved_addresses set.

**Pull request merged**